symfony/messenger/4.1/manifest.json

```diff
@@ -3,7 +3,22 @@
         "config/": "%CONFIG_DIR%/"
     },
     "env": {
-        "#1": "MESSENGER_TRANSPORT_DSN=amqp://guest:guest@localhost:5672/%2f/messages"
+        "#1": "MESSENGER_TRANSPORT_DSN=amqp://symfony:password@localhost:5672/%2f/messages"
+    },
+    "docker-compose": {
+        "services": [
+            "rabbitmq:",
+            "  image: rabbitmq:3.7.14",
+            "  environment:",
+            "    # You should definitely change the password in production",
+            "    - RABBITMQ_DEFAULT_USER=symfony",
+            "    - RABBITMQ_DEFAULT_PASS=password",
+            "  volumes:",
+            "    - rabbitmq-data:/var/lib/rabbitmq",
+            "  ports:",
+            "    - \"5672:5672\""
```

Comment:
Setting a localhost port makes it harder to have several projects running in parallel. I would only expose the container port and let Docker choose a random port for the localhost.

Reply:
Is it possible with Docker or Docker compose? AFAIK there are no built-in ways to do that, you have to rely on external tools such as https://github.com/weaveworks/weave.

Also, it will degrade the DX in some ways. For instance tools like MySQL Workbench ou PHPStorm DB expect the service to run on the default port.

Reply:
It works really well with Docker compose. Harcoding the port means that there is no way to run several projects in parallel on a laptop, that's a DX issue as well :)

Reply:
Do you have an example of the syntax?

Reply:
An idea, do not expose any port and suggest one to use `docker-compose.override.yml` file to customize his config like exposing a port.

Reply:
Random port forces the end user to change any env var related to the configuration urls in `.env.local`, especially if container is recreated with a new random port, because I'm not sure any Docker or Compose configuration could propagate the random value to any env var. I'm not sure it's the best solution for DX.

If anyone is working on multiple projects in parallel, my personal recommendation would be to `docker-compose down/stop` before leaving the previous project, and `docker-compose up -d` on the next project to work on, as a measure of safety for ports.

@B-Galati's idea is better IMO: no default port, and only an override one. This invites the recipe to have a `docker-compose.override.yaml.dist` file, and forces the installation workflow to copy/paste it to `docker-compose.override.yaml` at setup to make sure a port can be publicly used, to improve DX when setting up the project.

Reply:
The Symfony CLI is able to deal with random ports as Docker exposes the ports in its API. So, you always get the right env var for all your services and you are still able to run as many projects in parallel as you want. Best DX possible.